### ISSUE TITLE
Custom DXVK Config File Not Loading

### ISSUE THREAD

**Issue by sonic2kk** [sonic2kk/steamtinkerlaunch#492]
STL exports `DXVK_CONFIG_FILE` to the correct path and I have verified it finds it correctly if the path exists (and if it's uncommented from the game's configuration file). However it is not loaded with the game. I tested this by adding `dxvk.hud = fps` to the config file.

Interestingly, launching a game without STL and adding `DXVK_CONFIG_FILE="/home/user/.config/steamtinkerlaunch/dxvk/appid.conf" %command%` works fine. The DXVK wiki page [specifically notes](https://github.com/doitsujin/dxvk/wiki/Configuration#configuration-file) that this is an environment variable though.

Perhaps this should be part of the launch command?

**Comment by frostworx** [comment]
hm, very strange. it is exported so it should just work.
I'll add a writelog line, maybe it helps to find the issue.

**Comment by frostworx** [comment]
done :)

**Comment by sonic2kk** [comment]
Thanks, that log line outputs:
```
Tue Jun  7 19:30:19 BST 2022 INFO - prepareProton - exporting variable 'DXVK_CONFIG_FILE' pointing to '/home/user/.config/steamtinkerlaunch/dxvk/584400.conf
```

Not entirely sure why games aren't seeing it since that variable does work if a game is launched without STL.

**Comment by frostworx** [comment]
This "function" is really ancient, and I'm not sure if it really makes much sense it its current state.
might be a good opportunity to re-think it generally. tbh it looks pretty ugly :)

**Comment by frostworx** [comment]
> Thanks, that log line outputs:
> 
> ```
> Tue Jun  7 19:30:19 BST 2022 INFO - prepareProton - exporting variable 'DXVK_CONFIG_FILE' pointing to '/home/user/.config/steamtinkerlaunch/dxvk/584400.conf
> ```
> 
> Not entirely sure why games aren't seeing it since that variable does work if a game is launched without STL.

thanks for testing! strange indeed. need to look deeper into this apparently...

**Comment by frostworx** [comment]
hm strange, dumping the whole env directly before the game launch doesn't reveal anything useful as well.
the var exists and the path is correct.

**Comment by frostworx** [comment]
I'm not sure if this is really a blocker for #493. as already mentioned the `DXVK_CONFIG_FILE` variable is exported fine here.
will have to test a bit more soonish.
Do you have an example dxvk config which makes it very obvious that it was loaded? _(tbh I barely played with it yet generally)_

**Comment by sonic2kk** [comment]
Adding `dxvk.hud = full` to a DXVK config file should be the most obvious, it should display the entire DXVK hud so it's very visually clear that the config file is loaded. Just add that line to a DXVK config file at `/home/user/.config/steamtinkerlaunch/dxvk/appid.conf`, and make sure the DXVK config file is enabled for the game :) 

To test without STL, set the variable in the Steam launch options to `DXVK_CONFIG_FILE="/home/user/.config/steamtinkerlaunch/dxvk/appid.conf" %command%` and it should be loaded correctly, displaying the DXVK hud

**Comment by frostworx** [comment]
hehe, thanks for the detailed explanation 😀
will test a bit soon

**Comment by frostworx** [comment]
I can reproduce the issue, but it is very hard to debug, because the variable is exported correctly on game launch.
The only sane reason which comes into my mind currently is a different variable which disables `DXVK_CONFIG_FILE`
_(maybe worth to look into dxvk source code later)_

edit: found several interesting variables which differ, but hardcoding them didn't help. very weird issue...

**Comment by frostworx** [comment]
even when copying the found `$STLDXVKCFG` to `$PWD` _(i.e. the game dir)_
instead of exporting the variable `DXVK_CONFIG_FILE`
the config is ignored... very very weird issue...

**Comment by frostworx** [comment]
enabling dxvk logging reveals that the config behind `DXVK_CONFIG_FILE` is actually found by dxvk....
very very very weird issue.
enough for this weekend :)

**Comment by frostworx** [comment]
~~just compared the dxvk internal logs between steam launch option launch and steamtinkerlaunch internal launch:~~
~~- the steam launch option launch is missing a `VK_KHR_external_memory_win32` entry, but probably more important~~
~~- the logged DXVK version used is `v1.10.1-async` via steamtinkerlaunch and `DXVK: v1.10.1` via launch option _(the DXVK_ASYNC variable is 0 in steamtinkerlaunch though)_~~

~~edit: hmm, might be proton version related though: used latest official stable without steamtinkerlaunch (I don't use/have custom proton versions directly as compatibility tool) and latest GE with steamtinkerlaunch~~

edit: no difference between the logs when using proton-7.0-3 on both sides

**Comment by frostworx** [comment]
The issue is caused by your default `DXVK_HUD=0` 😀

**Comment by frostworx** [comment]
above commit fixes the issue _(tested and I don't see a reason to make it "better")_

**Comment by frostworx** [comment]
above commit makes the fix "better" 🤓

**Comment by sonic2kk** [comment]
Sorry for the delayed response, thank you for the extensive debugging on this issue! I just tested and this now works. I tested the DXVK HUD option and `d3d9.maxFrameRate = 30` (a config file specific option to limit only D3D9 games to a given framerate) and they both work now!

So the DXVK config file wasn't working because DXVK_HUD was 0? That is very interesting, or maybe it was just the DXVK hud itself being hidden (*I wonder what the order of precedence is for DXVK environment variables then*). My bad if this is related to the change I made before for the DXVK HUD :sweat_smile: 

Thank you for all the work though! :smile: Great that this is resolved
